fix: #232 disable login button when email or password fields are empty

Sanchez86 opened the pull request:
- Added disabled prop to AppButton that checks if email or password are empty
- Login button now only becomes clickable when both fields contain data
- Prevents form submission with incomplete credentials
- Resolves issue where login button was clickable without entering any data

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

* **Bug Fixes**
  * The login submit button is now disabled until both email and password are provided and the email passes validation, preventing accidental submissions with incomplete or invalid credentials.
  * The button continues to stay disabled during loading to avoid duplicate submissions.
  * Improves clarity and reduces errors during sign-in by providing a consistent disabled state and clearer user feedback when inputs are missing or invalid.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Sanchez86 committed:
fix: #232 disable login button when email or password fields are empty

- Added disabled prop to AppButton that checks if email or password are empty
- Login button now only becomes clickable when both fields contain data
- Prevents form submission with incomplete credentials
- Resolves issue where login button was clickable without entering any data

AndiKot commented on src/containers/guest-home-page/login-form/LoginForm.jsx:
```diff
@@ -70,7 +70,12 @@ const LoginForm = ({
 
-      <AppButton loading={authLoading} sx={styles.loginButton} type='submit'>
+      <AppButton
+        disabled={!data.email || !data.password}
```
It must be disabled when incorrect email entered too

Sanchez86 committed:
fix: disable login button for invalid email format

- Add validation check for incorrect email format in disabled prop
- Button now disabled when: empty fields OR invalid email format
- Improves UX by preventing submission with invalid email
- Uses existing email validation from useForm errors object

Addresses feedback for task #232
